Week8/app/src/main/java/com/em4n0101/mytvshows/ui/searchshow/SearchShowFragment.kt

```diff
@@ -0,0 +1,144 @@
+package com.em4n0101.mytvshows.ui.searchshow
+
+import android.content.Intent
+import android.content.res.Configuration
+import android.net.ConnectivityManager
+import android.os.Bundle
+import android.view.*
+import android.widget.EditText
+import androidx.appcompat.widget.SearchView
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.GridLayoutManager
+import com.em4n0101.mytvshows.MyTvShowsApplication
+import com.em4n0101.mytvshows.R
+import com.em4n0101.mytvshows.model.Success
+import com.em4n0101.mytvshows.model.Show
+import com.em4n0101.mytvshows.networking.NetworkingStatusChecker
+import com.em4n0101.mytvshows.ui.showdetail.ShowDetailActivity
+import com.em4n0101.mytvshows.utils.toast
+import kotlinx.android.synthetic.main.fragment_search_show.*
+import kotlinx.coroutines.launch
+
+class SearchShowFragment : Fragment() {
+
+    private val remoteApi = MyTvShowsApplication.remoteApi
+    private val networkStatusChecker by lazy {
+        NetworkingStatusChecker(activity?.getSystemService(ConnectivityManager::class.java))
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_search_show, container, false)
+    }
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        activity?.let {
+            // setup recycler
+            if (context?.resources?.configuration?.orientation == Configuration.ORIENTATION_PORTRAIT) {
+                showsRecyclerView.layoutManager = GridLayoutManager(context, 2)
+            } else {
+                showsRecyclerView.layoutManager = GridLayoutManager(context, 4)
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.search_menu, menu)
+        val searchItem = menu.findItem(R.id.actionSearchShow)
+        if (searchItem != null) {
+            val searchView = searchItem.actionView as SearchView
+            val editText = searchView.findViewById<EditText>(R.id.search_src_text)
+            editText.hint = getString(R.string.hintSearchShow)
+
+            searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+                override fun onQueryTextSubmit(query: String?): Boolean = handlerOnQueryTextSubmit(query)
+
+                override fun onQueryTextChange(newText: String?): Boolean = handlerOnQueryTextChange(newText)
+            })
+        }
```

Comment:
I would extract this into another method as well. If you think about it, the method is called `onCreateOptionsMenu` but you are handling searchView input and text changing here as well.